lib/plottr_components/src/components/dashboard/account/proOnboarding/ProStep2.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'react-proptypes'
+import { t } from 'plottr_locales'
+import OnboardingStep from '../../../onboarding/OnboardingStep'
+import { StepBody, StepHeader } from '../../../onboarding/Step'
+import UnconnectedUploading from './Uploading'
+
+// is there a way to know that this step is done?
+
+const isPlottrCloudFile = (filePath) => filePath && filePath.startsWith('plottr://')
+// needs to be declared here otherwise it will cause an infinite loop since we'd be passing in a new [] every render
+const initialFrbFiles = []
+
+const ProStep2Connector = (connector) => {
+  const {
+    platform: {
+      file: { useSortedKnownFiles },
+      template: { useCustomTemplatesInfo },
+    },
+  } = connector
+
+  const Uploading = UnconnectedUploading(connector)
+
+  const ProStep2 = ({ nextStep }) => {
+    const [sortedIds, filesById] = useSortedKnownFiles('', initialFrbFiles)
+    const [templateInfo] = useCustomTemplatesInfo()
+    const [nothing, setNothing] = useState(false)
+    const [view, setView] = useState('choice')
+    const [templates, setTemplates] = useState(null)
+    const [projects, setProjects] = useState(null)
+
+    useEffect(() => {
+      // TODO: figure out what to do with templates/projects already in firebase
+
+      // discover all templates
+      const templatesList = Object.values(templateInfo)
+      if (!templates && templatesList.length) {
+        setTemplates(templatesList)
+      }
+      // discover all projects
+      if (!projects && sortedIds.length) {
+        const projectsToUpload = sortedIds.reduce((acc, id) => {
+          const project = filesById[id]
+          if (!isPlottrCloudFile(project.path)) {
+            acc.push(project)
+          }
+          return acc
+        }, [])
+        setProjects(projectsToUpload)
+      }
+
+      if (!templatesList.length && !sortedIds.length) {
+        setNothing(true)
+      } else {
+        setNothing(false)
+      }
+    }, [sortedIds, templateInfo])
+
+    useEffect(() => {
+      // there's nothing to upload
+      if (templateInfo && sortedIds && nothing) nextStep()
+    }, [nothing])
+
+    if (!templates && !projects && !nothing) {
```

Comment:
`!nothing` feels confusing and might cause a glitch in rendering.
If we initialised it to `null` then we'd have the semantics down very precisely to "when stuff hasn't yet loaded".